src/apps/vpn/mozillavpn.cpp

```diff
@@ -755,14 +765,11 @@ void MozillaVPN::accountChecked(const QByteArray& json) {
   m_private->m_user.writeSettings();
   m_private->m_deviceModel.writeSettings();
 
-  if (m_private->m_user.subscriptionNeeded() && state() == StateMain) {
-    NotificationHandler::instance()->subscriptionNotFoundNotification();
-    maybeStateMain();
+  if (!m_private->m_user.subscriptionNeeded() || state() != StateMain) {
     return;
   }
-
-  // To test the subscription needed view, comment out this line:
-  // m_private->m_controller.subscriptionNeeded();
+  NotificationHandler::instance()->subscriptionNotFoundNotification();
```

Comment:
Also this can go to ProductsHandler. You can monitor the use subscription state and show the notification

Reply:
What about this?